### PR TITLE
Added Profile and Endpoints for Stock Interactions

### PR DESCRIPTION
**Pull request by javidangarcia**
- Added profile component that is accessible by clicking on a user's name.
- Added full name and profile pictures for users in database.
- Added GET endpoint for /user/:username that retrieves that user's information to display on profile.
- Added profile history component that displays user stock interactions on profile view.
- Added GET endpoints for /follows, /likes, and /dislikes that retrieves stock interactions based on username.

https://github.com/javidangarcia/stock-zone/assets/89328228/dbe96582-4482-40c7-aa65-9e34def68d0e

### CODE REVIEW

File: backend/routes/users.js [hunk 1]
```diff
@@ -2,18 +2,44 @@ import express from "express";
...
+            where: { username }
+        });
+    
+        if (user === null) {
+            res.status(404).json({ error: "This user doesn't exist." });
+            return;
+        }
+    
+        res.status(200).json({
+            user: {
```
Comment (BriptimusPrimus) [inline backend/routes/users.js:23]: Good job avoiding sending the password in the response :)

File: backend/routes/users.js [hunk 2]
```diff
@@ -2,18 +2,44 @@ import express from "express";
 import { User } from "../models/user.js";
 import { Op } from "sequelize";
 import bcrypt from "bcrypt";
+import { PROFILE_PICTURE } from "../utils.js";
```
Comment (BriptimusPrimus) [inline backend/routes/users.js:5]: 👍

File: backend/server.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ import commentRoutes from "./routes/comments.js";
 import session from "express-session";
 import SequelizeStoreInit from 'connect-session-sequelize';
 import cookieParser from "cookie-parser";
+import { checkSession } from "./utils.js";
```
Comment (BriptimusPrimus) [inline backend/server.js:14]: 👍

File: frontend/src/components/Comments/Comments.jsx [hunk 4]
```diff
@@ -75,9 +76,19 @@ export default function Comments({ ticker }) {
...
-                            <p className="comment-date">{formatDateTime(comment.createdAt)}</p>
-                            <p className="comment-content">{comment.content}</p>
+                            <div className="comment-picture">
+                                <img
+                                    src={comment.User.picture}
+                                    alt={`This is a profile picture associated with ${comment.User.username}`}
+                                />
+                            </div>
+                            <div className="comment-content">
+                                <Link to={`/profile/${comment.User.username}`} className="user-link">
```
Comment (BriptimusPrimus) [inline frontend/src/components/Comments/Comments.jsx:86]: It's better not to assume that the comment object has a User object, that has a username prop. 
The safer way is `comment?.User?.username ?? ''`

Comment (BriptimusPrimus) [reply frontend/src/components/Comments/Comments.jsx:86]: Nice Link 👍

File: frontend/src/components/Navbar/Navbar.jsx [hunk 5]
```diff
@@ -12,16 +12,19 @@ export default function Navbar() {
...
-            const response = await axios.post("http://localhost:3000/users/logout", null, {
-                withCredentials: true
-            });
+            const response = await axios.post(
+                "http://localhost:3000/users/logout",
+                null,
+                {
+                    withCredentials: true
+                }
+            );
```
Comment (BriptimusPrimus) [inline frontend/src/components/Navbar/Navbar.jsx:21]: Was this the result of running prettier?

File: frontend/src/components/Navbar/Navbar.jsx [hunk 6]
```diff
@@ -38,7 +41,15 @@ export default function Navbar() {
             <div className="nav-profile">
                 {user ? (
                     <div className="logout">
```
Comment (BriptimusPrimus) [inline frontend/src/components/Navbar/Navbar.jsx:43]: Super Nit: After this change, it seems that "logout" is not the most accurate class name anymore. 
The DIV became a mini profile data with a logout link next to it.

File: frontend/src/utils.js [hunk 7]
```diff
@@ -6,8 +6,15 @@ export const formatDate = (dateString) => {
...
-    return word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();
+export const capitalize = (sentence) => {
+
+    const words = sentence.split(" ");
+
+    const capitalizedWords = words.map((word) => {
+        return word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();
+    });
+
+    return capitalizedWords.join(" ");
```
Comment (BriptimusPrimus) [inline frontend/src/utils.js:17]: Good job writing this function. 

One thing to notice is that every function you are using here is non mutating, and returns a new object, so they can be easily chained:

```
return sentence.split(" ")
  .words.map((word) => {
    return word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();
  })
  .join(" ");
```

It's the same thing, but without capturing intermediate results in variables. In many cases you don't need variables. I want to know your thoughts. What option seems better to you, and why?

File: frontend/src/components/Navbar/Navbar.jsx (hunk 5 above)
Comment (javidangarcia) [reply frontend/src/components/Navbar/Navbar.jsx:21]: Yeah, sometimes prettier makes my code look weird.

File: frontend/src/components/Comments/Comments.jsx (hunk 4 above)
Comment (javidangarcia) [reply frontend/src/components/Comments/Comments.jsx:86]: It always has a User object though.

File: frontend/src/utils.js (hunk 7 above)
Comment (javidangarcia) [reply frontend/src/utils.js:17]: I like how the first one looks better and it's more organized to me and easier to read.

File: frontend/src/components/Navbar/Navbar.jsx (hunk 5 above)
Comment (BriptimusPrimus) [reply frontend/src/components/Navbar/Navbar.jsx:21]: In prettier's defense I will say that in my opinion, the code looks better after it runs.

**Pull request merged**